File: modules/auth/app/jp/co/recruitms/auth/controllers/AuthenticateController.java
package jp.co.recruitms.auth.controllers;

import jp.co.recruitms.auth.security.Authentication;
import jp.co.recruitms.auth.security.PlayAuthenticator;
import jp.co.recruitms.database.models.User;
import jp.co.recruitms.database.db.Users;
import play.mvc.*;

public class AuthenticateController extends Controller {
  public Result signIn(String userName, String pass) {
    if (userExists(userName, pass)) {
     return ok(userName);
    } else {
      return unauthorized("User not found.");
    }
  }

  public Result signUp(String userName, String pass) {
    if (!userExists(userName, pass)){
      return Users
        .save(new User(userName, pass))
        .map(user -> ok(user.name))
        .orElseGet(() -> internalServerError("Create User failed"));
    } else {
     return badRequest("User already exists.");
    }
  }

  @Security.Authenticated(PlayAuthenticator.class)
  public Result logout(Http.Request req){
    return Authentication.getUsername(req)
      .map(Results::ok)
      .orElseGet(() ->badRequest("Not logged in."));
  }

  private boolean userExists(String userName, String password){
    return Users.findByName(userName).map(user -> user.password.equals(password)).orElse(false);
  }
}

File: modules/auth/app/jp/co/recruitms/auth/controllers/AuthenticateController.scala
package jp.co.recruitms.auth.controllers

import javax.inject.Inject
import javax.inject.Singleton
import jp.co.recruitms.auth.controllers.actions.UserAction
import jp.co.recruitms.auth.controllers.actions.UserRequest
import jp.co.recruitms.auth.models.User
import jp.co.recruitms.auth.models.Users
import play.api.mvc.Action
import play.api.mvc.AnyContent
import play.api.mvc.BaseController
import play.api.mvc.ControllerComponents
import play.api.mvc.Request

@Singleton
class AuthenticateController @Inject() (val controllerComponents: ControllerComponents, userAction: UserAction)
    extends BaseController {
  def signIn(userName: String, pass: String): Action[AnyContent] = Action { implicit req: Request[AnyContent] =>
    if (userExists(userName, pass))
      Ok(userName)
    else
      Unauthorized("User not found.")
  }

  def signUp(userName: String, pass: String): Action[AnyContent] = Action { implicit req: Request[AnyContent] =>
    if (!userExists(userName, pass))
      Users
        .save(User(userName, pass))
        .map(user => Ok(user.name))
        .getOrElse(InternalServerError("Create User failed"))
    else
      BadRequest("User already exists.")
  }

  def logout(): Action[AnyContent] = userAction { implicit req: UserRequest[AnyContent] =>
    req.userName
      .map(name => {
        Ok(name)
      })
      .getOrElse(BadRequest("Not logged in."))
  }

  private def userExists(userName: String, password: String): Boolean =
    Users.findByName(userName).exists(_.password == password)
}
